Lib/typing.py

```diff
@@ -1828,8 +1828,13 @@ def __subclasscheck__(cls, other):
                 not cls.__callable_proto_members_only__
                 and cls.__dict__.get("__subclasshook__") is _proto_hook
             ):
+                non_method_attrs = {
```

Comment:
Using a set comprehension here means the test is probably going to randomly fail 50% of the time, because we don't know what the order is going to be when we call `str()` on the set. Use `sorted()` with a generator expression instead.

Reply:
Fixed it. One thing I have been wondering though: if we are talking about non-method members, shouldn't we be testing using `inspect.ismethod` (which checks `isinstance(object, types.MethodType)`)?

Reply:
I guess that wouldn't catch classmethods...

Reply:
No, because we're accessing the method from the class itself, not an instance of the class:

```pycon
>>> class Foo:
...   def bar(self): pass
...   
>>> import inspect
>>> inspect.ismethod(Foo().bar)
True
>>> inspect.ismethod(Foo.bar)
False
```